### PR TITLE
Fix crash using binary reader logging with error

### PR DESCRIPTION
**Pull request by binji**
When the binary logging is turned on, it can't use the
`reader->on_error` directly since that function expects that the
user_data is the BinaryReaderInterpreter context (for example), but it
is actually the LoggingContext.

Some additional changes:

* Add a test that uses logging and succeeds
* Add a test that uses logging and has an error
* Change the `run-*-interp.py` scripts to use `-t` for tracing and `-v`
  for logging

### CODE REVIEW

File: src/binary-reader.cc [hunk 1]
```diff
@@ -2028,7 +2039,7 @@ Result read_binary(const void* data,
   WABT_ZERO_MEMORY(logging_reader);
   logging_reader.user_data = &logging_context;
 
-  logging_reader.on_error = reader->on_error;
+  logging_reader.on_error = logging_on_error;
```
Comment (sbc100) [inline src/binary-reader.cc]: The reason I did this is because some tools don't register an on_error handler and what to default to default one.   If logging is enabled the default handler is not used, so the error gets swallowed.  Try wasmdump or wasm-link with --debug on a file with an error to see this.

Comment (binji) [inline src/binary-reader.cc]: OK, I've reworked the on_error handlers to return a bool which says whether the error was handled.

File: CMakeLists.txt [hunk 2]
```diff
@@ -106,10 +106,6 @@ else ()
     add_definitions(-Wno-clobbered)
   endif ()
 
-  if (COMPILER_IS_CLANG)
-    add_definitions(-fcolor-diagnostics)
-  endif ()
```
Comment (sbc100) [inline CMakeLists.txt]: Not part of this change?

Comment (binji) [inline CMakeLists.txt]: Whoops, done
